js_modules/dagit/packages/core/src/instance/backfill/BackfillPage.tsx

```diff
@@ -184,31 +184,75 @@ export const BackfillPage = () => {
           </thead>
           <tbody>
             {backfill.assetBackfillData?.assetBackfillStatuses.map((asset) => {
-              return asset.__typename === 'AssetPartitionsStatusCounts' ? (
-                <tr key={asset.assetKey.path.join('/')}>
-                  <td>
-                    <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
-                      <div>
-                        <a href={assetDetailsPathForKey(asset.assetKey)}>
-                          {asset.assetKey.path.join('/')}
-                        </a>
-                      </div>
-                      <div>
-                        <StatusBar
-                          targeted={asset.numPartitionsTargeted}
-                          requested={asset.numPartitionsInProgress}
-                          completed={asset.numPartitionsMaterialized}
-                          failed={asset.numPartitionsFailed}
-                        />
-                      </div>
-                    </Box>
-                  </td>
-                  <td>{asset.numPartitionsTargeted}</td>
-                  <td>{asset.numPartitionsInProgress}</td>
-                  <td>{asset.numPartitionsMaterialized}</td>
-                  <td>{asset.numPartitionsFailed}</td>
-                </tr>
-              ) : null;
+              if (
+                ['AssetPartitionsStatusCounts', 'UnpartitionedAssetStatus'].includes(
+                  asset.__typename,
+                )
+              ) {
```

Comment:
`return null` early to unindent and make it a little easier to follow?

Reply:
This actually feels like a GraphQL error -- was this an array of optionals because of the unpartitioned assets? I guess I don't see a third case that this is omitting anymore -- it's a union with just these two types in it, maybe we can tweak the schema so this check isn't necessary?